src/internet_identity/src/anchor_management/tentative_device_registration.rs

```diff
@@ -86,15 +89,16 @@ pub async fn add_tentative_device(
 }
 
 pub fn verify_tentative_device(
+    anchor: &mut Anchor,
     anchor_number: AnchorNumber,
     user_verification_code: DeviceVerificationCode,
-) -> VerifyTentativeDeviceResponse {
+) -> Result<(VerifyTentativeDeviceResponse, Operation), VerifyTentativeDeviceResponse> {
```

Comment:
Returning `Result<(R, ...), R>` looks like an unusual pattern -- maybe comment this function, especially what is semantics of the returned `R` in both cases?  Or maybe this pattern is not necessary?

Reply:
I agree, I'm not quite happy with the interface either. The reason we get `Result<R,R>` is basically that we have variants on the `candid` interface which unify success and error conditions on the same type. E.g.:
```
type RegisterResponse = variant {
    // A new user was successfully registered.
    registered: record {
        user_number: UserNumber;
    };
    // No more registrations are possible in this instance of the II service canister.
    canister_full;
    // The challenge was not successful.
    bad_challenge;
};
```
Also, the `authenticated_anchor_operation` requires information about an operation to do bookkeeping and archiving.


So, we could go with `(R, Option<Operation>)`, but this would then (from a typing perspective) indicate that the `Operation` is entirely optional (which it is not).

Or we could split the interface to `Result<R,E>` and only assemble the final response type outside of the `authenticated_anchor_operation`. I think, this would be the "nice" solution, but introduce quite some additional internal types. So essentially, the current design is a compromise between having accurate types and still being able to communicate success / failure to the caller.

Reply:
As discussed, I agree that the "nice" solution would be probably an overkill, so leaving the interface as-is SGTM.  Maybe refer in a comment to the documentation of `authenticated_anchor_operation`, so that it is clear why the `Result...` has this structure?